client/src/api/v3/boost.ts

```diff
@@ -1,28 +1,121 @@
 import { emit } from 'api/common/socket';
-import { BoostConfigType } from 'types/boost';
+import {
+  FileConfigT,
+  ConfigBundleT,
+  ConfigObjT,
+  ConfigT,
+  fileConfigTypeToRuntype,
+} from 'types/boost';
+import toast from 'react-hot-toast';
+import { Runtype } from 'runtypes';
+
+type payloadAction = 'upload' | 'delete';
+
+/**
+ * Check that the given content is of the type given
+ *
+ * @param content the content to check for it's type
+ * @param typeRequired the type the content is expected to be
+ *
+ * @returns true if the content satisfies the given type else false
+ */
+function isCorrectContentType(content: any, typeRequired: Runtype) {
+  try {
+    if (typeRequired.check(content)) {
+      return true;
+    }
+  } catch (error) {
+    return false;
+  }
+  return false;
+}
 
 /**
- * Send the content of the given configuration file on `boost/configs/action` over MQTT
+ * Send configuration status over MQTT on topic 'boost/configs/action'
  *
+ * @param actionType represents whether the config is being uploaded or deleted
  * @param type the type of the configuration being sent
- * @param configFile file containing the configuration content
+ * @param name name of the config file
+ * @param configContent configuration content
+ */
+function sendConfig(
```

Comment:
This function takes in parameters, renames them, and then emits the object as a JSON string.
Perhaps this could be refactored to take a payload and so type checks on the content (i.e. passing in the content as an object rather than string).
This would also reduce the code complexity later on.

Reply:
Sorry I'm not completely following on here, what do you mean by 'rename'. I think you're referring to taking in the `configContent` as an object rather than a `string`, and I kind of do agree with that. Let me know if I have not got you correct on this one :)

Reply:
[These lines](https://github.com/monash-human-power/dashboard/pull/105/files/5c21ee1ce9cec598347dac2996a608f22f495d00#diff-b1c06b76b608351d0758ef2d52f62dc9ed9bc650aff39c5d45f8370beafbace4R48-R53) essentially take the inputs (`actionType, type, name, configContent`) and then renames (`action, configType, fileName, content`), which seems a bit redundent to me. So really considering what useful thing this function does, it's really only the `stringify` and `emit` which is a single line anyway. 
My thought was that the RunType checks could be added here to reduce the type checking code later on.

Reply:
Ahh I understand now, the reason why it's an abstracted function is because later on the feature to delete a config file can utilise this function as the payload would be the same.

Reply:
I wasn't questioning why it's abstracted, but that it could be doing more useful things than renaming parameters (like checking the run types).

Reply:
Imho, I don't think it really needs to do the checking. It's functionality is to handle sending the uploaded or deleted config over MQTT and so it's reasonable to expect the correct content to be given as parameters. So it's fine I believe if that means it's a small modular func. The renaming (to be literal) is only there because the names can't be the same otherwise the compiler complains. As for making the checking part, I believe it's right to do it in `uploadConfig` because that's where you check whether the upload is of the right type. Sorry if I'm still getting you wrong on this one 😂.

Reply:
In that case, `sendConfig` does nothing other than call `emit` with its parameters, so it could just be replaced directly with `emit`.
I am aware that the renaming is something that is forced, but I think we would agree that it's not particularly useful. So considering only useful things that the function does, it is only sending the data, which is already implemented by `emit`.